docs/contributing/making-a-pull-request.md

```diff
@@ -0,0 +1,450 @@
+
+- Ideally, also test the examples in the [examples repository]. If you are
+working on the Insiders edition of Material for MkDocs, you can simply start a
+build at the top level and the [projects plugin] will build all of the examples
+for you. If you are on the public version, you will need to build each
+sub-project individually. We appreciate that this is a growing collection of
+examples and you may want to prioritize those that are most relevant to the
+functionality you change.
+
+[examples repository]: https://github.com/mkdocs-material/examples
+[projects plugin]: https://squidfunk.github.io/mkdocs-material/plugins/projects/
+
+### Creating the pull request
+
+Initially, create the pull request **as a draft**. You do this [through the
+various interfaces that GitHub provides]. Which one you use is entirely up to
+you. We do not provide specific instructions for using the interfaces as GitHub
+provide all the information that should be necessary.
+
+[through the various interfaces that GitHub provides]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
+
+!!! question "Pull Request Template?"
+
+    @squidfunk, would it be useful to have a [PR template]? That might replace a
+    bunch of paragraphs of guidance we would otherwise need to add to this
+    document?
+
+[PR template]: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository
+
+### Commits, messages, mistakes and 'squash'
+
+### Deleting branches
+
+Once the pull request has been merged into the master branch of the Material
+for MkDocs repository, you should remove the branch both from the fork on
+GitHub and from the local clone on your computer. This avoids possible
+confusion about the state of development.
+
+First, switch back to the `master` branch with `git switch master` and then
+delete the branch used for the PR using `git branch -d <name>`.
+
+### Subsequent Pull Requests
+
+It is important that subsequent pull requests are started from an up-to-date
+history of the `master` branch. One way to achieve this is to delete the fork
+and start with an entirely new one next time round.
+
+If you contribute to Material for MkDocs more often or just happen to be
+doing two or more pull requests in succession, you can also just make sure
+to sync your fork (using the GitHub UI) and pull from it into your local
+repository. So, just delete the topic branch you created (both locally and in
+your fork) and pull from the main repository's `master` branch into your
+`master` branch before starting work on a new pull request.
+
+## Dos and Don'ts
+
+1. **Do** discuss what you intend to do with people in the discussions so that the
+   rational for any changes is clear before you write or modify code.
+
+2. **Don't** just create a pull request with changes that are not explained.
```

Comment:
I'd move this to the top, as it's the only "Don't" and most important thing of all.